src/dmd/backend/evalu8.d

```diff
@@ -82,9 +89,7 @@ int boolres(elem *e)
                 case TYbool:
                 case TYwchar_t:
                 case TYenum:
-#if !MARS
                 case TYmemptr:
-#endif
```

Comment:
Shouldn't this result in a `version(MARS){} else { ... }` ?

Reply:
I decided that having too many version statements was clutter, and so removed the ones that weren't really necessary.